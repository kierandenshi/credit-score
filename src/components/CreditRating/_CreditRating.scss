@import '~@css/includes/helpers';

.credit-rating {
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  width: 298px;
  height: 298px;

  &__info {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  @include media('>phone') {
    width: 414px;
    height: 414px;
  }
}
