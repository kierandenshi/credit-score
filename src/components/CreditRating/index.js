export { default } from './CreditRating.jsx';
