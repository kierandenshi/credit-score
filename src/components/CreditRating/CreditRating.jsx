import React from 'react';
import PropTypes from 'prop-types';
import RadialChart from '@components/RadialChart';
import './_CreditRating.scss';

const CreditRating = ({ data }) => {
    const { minScoreValue, maxScoreValue, score } = data;

    return (
        <div className={'credit-rating'}>
            <div className={'credit-rating__chart'}>
                <RadialChart
                    minValue={minScoreValue}
                    maxValue={maxScoreValue}
                    value={score}
                />
            </div>
            <div className={'credit-rating__info'}>
                <span>Your credit rating is</span>
                <span className={'strong'}>{score}</span>
                <span>{`out of ${maxScoreValue}`}</span>
            </div>
        </div>
    );
};

CreditRating.propTypes = {
    data: PropTypes.shape({}).isRequired,
};

export default CreditRating;
