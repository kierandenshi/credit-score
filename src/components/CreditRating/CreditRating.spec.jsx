import React from 'react';
import { shallow } from 'enzyme';
import { getProps } from '@test/helpers';
import CreditRating from './CreditRating.jsx';

const initialProps = {
    data: { minScoreValue: 0, maxScoreValue: 700, score: 450 },
};

describe('CreditRating component', () => {
    const wrapper = shallow(<CreditRating {...getProps(initialProps)} />);

    it('should render', () => {
        expect(wrapper.find('.credit-rating')).toHaveLength(1);
    });
});
