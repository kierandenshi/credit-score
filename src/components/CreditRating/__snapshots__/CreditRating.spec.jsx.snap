// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditRating component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): ShallowWrapper {
    "length": 1,
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <CreditRating
      data={
        Object {
          "maxScoreValue": 700,
          "minScoreValue": 0,
          "score": 450,
        }
      }
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="credit-rating__chart"
          >
            <RadialChart
              maxValue={700}
              minValue={0}
              value={450}
            />
          </div>,
          <div
            className="credit-rating__info"
          >
            <span>
              Your credit rating is
            </span>
            <span
              className="strong"
            >
              450
            </span>
            <span>
              out of 700
            </span>
          </div>,
        ],
        "className": "credit-rating",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RadialChart
              maxValue={700}
              minValue={0}
              value={450}
            />,
            "className": "credit-rating__chart",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "maxValue": 700,
              "minValue": 0,
              "value": 450,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Your credit rating is
              </span>,
              <span
                className="strong"
              >
                450
              </span>,
              <span>
                out of 700
              </span>,
            ],
            "className": "credit-rating__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your credit rating is",
              },
              "ref": null,
              "rendered": "Your credit rating is",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 450,
                "className": "strong",
              },
              "ref": null,
              "rendered": 450,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "out of 700",
              },
              "ref": null,
              "rendered": "out of 700",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="credit-rating__chart"
            >
              <RadialChart
                maxValue={700}
                minValue={0}
                value={450}
              />
            </div>,
            <div
              className="credit-rating__info"
            >
              <span>
                Your credit rating is
              </span>
              <span
                className="strong"
              >
                450
              </span>
              <span>
                out of 700
              </span>
            </div>,
          ],
          "className": "credit-rating",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <RadialChart
                maxValue={700}
                minValue={0}
                value={450}
              />,
              "className": "credit-rating__chart",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": null,
                "maxValue": 700,
                "minValue": 0,
                "value": 450,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Your credit rating is
                </span>,
                <span
                  className="strong"
                >
                  450
                </span>,
                <span>
                  out of 700
                </span>,
              ],
              "className": "credit-rating__info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Your credit rating is",
                },
                "ref": null,
                "rendered": "Your credit rating is",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 450,
                  "className": "strong",
                },
                "ref": null,
                "rendered": 450,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "out of 700",
                },
                "ref": null,
                "rendered": "out of 700",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="credit-rating__chart"
        >
          <RadialChart
            maxValue={700}
            minValue={0}
            value={450}
          />
        </div>,
        <div
          className="credit-rating__info"
        >
          <span>
            Your credit rating is
          </span>
          <span
            className="strong"
          >
            450
          </span>
          <span>
            out of 700
          </span>
        </div>,
      ],
      "className": "credit-rating",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <RadialChart
            maxValue={700}
            minValue={0}
            value={450}
          />,
          "className": "credit-rating__chart",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": null,
            "maxValue": 700,
            "minValue": 0,
            "value": 450,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Your credit rating is
            </span>,
            <span
              className="strong"
            >
              450
            </span>,
            <span>
              out of 700
            </span>,
          ],
          "className": "credit-rating__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your credit rating is",
            },
            "ref": null,
            "rendered": "Your credit rating is",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 450,
              "className": "strong",
            },
            "ref": null,
            "rendered": 450,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "out of 700",
            },
            "ref": null,
            "rendered": "out of 700",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="credit-rating__chart"
          >
            <RadialChart
              maxValue={700}
              minValue={0}
              value={450}
            />
          </div>,
          <div
            className="credit-rating__info"
          >
            <span>
              Your credit rating is
            </span>
            <span
              className="strong"
            >
              450
            </span>
            <span>
              out of 700
            </span>
          </div>,
        ],
        "className": "credit-rating",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RadialChart
              maxValue={700}
              minValue={0}
              value={450}
            />,
            "className": "credit-rating__chart",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "maxValue": 700,
              "minValue": 0,
              "value": 450,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Your credit rating is
              </span>,
              <span
                className="strong"
              >
                450
              </span>,
              <span>
                out of 700
              </span>,
            ],
            "className": "credit-rating__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your credit rating is",
              },
              "ref": null,
              "rendered": "Your credit rating is",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 450,
                "className": "strong",
              },
              "ref": null,
              "rendered": 450,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "out of 700",
              },
              "ref": null,
              "rendered": "out of 700",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
